MetaMask/metamask-mobile
# feat: disable wallet buttons for accounts that cannot sign transactions

## **Description**

This PR disables certain buttons in the `WalletActions` component when the selected account cannot sign transactions. This change improves the user experience by preventing attempts to perform actions that require transaction signing when the account cannot do so.

1. The reason for the change is to prevent users from attempting actions that their current account cannot perform.
2. The improvement is that Buy, Sell, Send, Swap, and Bridge buttons are now disabled when the selected account cannot sign transactions, providing clear visual feedback to the user about available actions.

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/570

## **Manual testing steps**

1. Go to the [SSK](https://metamask.github.io/snap-simple-keyring/latest/) and create an account
2. Update the account using this JSON, replacing the address and ID fields with the new account information (this update step removes `eth_signTransaction` from the account's methods)
```json
{
    "id": "your new account ID",
    "address": "you new account address",
    "options": {},
    "methods": [
        "personal_sign",
        "eth_sign",
        "eth_signTypedData_v1",
        "eth_signTypedData_v3",
        "eth_signTypedData_v4"
    ],
    "type": "eip155:eoa"
}
```
3. Switch to the account that cannot sign transactions if not already there
4. Verify that the Buy, Sell, Send, Swap, and Bridge buttons are visually disabled with a message and not clickable
5. Click on the asset details view
6. Verify that the Buy, Sell, Send, Swap, and Bridge buttons are visually disabled and not clickable
7. Switch back to an account that can sign transactions
8. Verify that the buttons are now enabled and functional in wallet view and asset details view

## **Screenshots/Recordings**

### **Before**

Wallet view
<img src="https://github.com/user-attachments/assets/d0cdcd63-cec5-4641-95a4-a47450319282" width=350 /> 

Asset details view
![simulator_screenshot_5FB98682-1B94-440A-9E12-BDE6D8B2B1C3](https://github.com/user-attachments/assets/ceac95af-3653-4178-9718-f75ff97e239f)


### **After**

Wallet view
![Simulator Screenshot - iPhone 15 Pro - 2024-10-24 at 15 43 57](https://github.com/user-attachments/assets/b3a4ed7c-3a79-413b-9ce2-6b05a5683de0)

Asset details view
![Simulator Screenshot - iPhone 15 Pro - 2024-10-24 at 15 44 06](https://github.com/user-attachments/assets/8962ef77-37c8-4bad-8d82-559dc1a808ce)


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
